New Loop Benchmarks

1933131 opened the pull request:
<!--
  New Loop Benchmarks having scientific computation. Adapted from C codes, heavily used in filter designs and approximations for Digital Signal Processing.
-->

- [x] Loop-scientific-comp under "Loops subcatgory of Reachsafety."
- [x] Apache 2.0 License. 
- [x] Details present in README.txt in Loop-scientific-comp directory. 

- [x] Added in Reachsafety-loops.set 
- [x] Reachsafety.prp
- [x] File names are given accordingly.

<!-- For C programs: -->
- [x] 32 bit compilation.
- [x] original sources present.
- [x] Preprocessed files present with required specifications.
- [x] preprocessed files generated with 32 bit architecture.
- [x] Makefile added.

kfriedberger commented on c/loop-scientific-comp/LICENSE.txt:
```diff
@@ -0,0 +1,202 @@
+
```
Replace identical license files with symlinks? see `c/LICENSE.Apache-2.0.txt`

kfriedberger commented on c/loop-scientific-comp/loop1_false-unreach-call.c:
```diff
@@ -0,0 +1,34 @@
+	{
+		term = term * (x/count) ; 
+		exp = exp + term ;
+		count++ ;
```
integer overflow? better use `unsigned int`.

kfriedberger commented on c/loop-scientific-comp/loop1_true-unreach-call.c:
```diff
@@ -0,0 +1,36 @@
+	{
+		term = term * (x/count) ; 
+		exp = exp + term ;
+		count++ ;
```
integer overflow? better use unsigned int.

kfriedberger commented on c/loop-scientific-comp/loop2_false-unreach-call.c:
```diff
@@ -0,0 +1,46 @@
+		
+		oddExp = oddExp + multFactor*term ;
+		
+		count++ ;
```
integer overflow? better use unsigned int.

kfriedberger commented on c/loop-scientific-comp/loop2_true-unreach-call.c:
```diff
@@ -0,0 +1,46 @@
+		
+		oddExp = oddExp + multFactor*term ;
+		
+		count++ ;
```
integer overflow? better use unsigned int.

dbeyer commented:
@1933131: Could you please make the changes proposed by @kfriedberger? I would like to merge this soon.

danieldietsch commented:
What about #581? I also think these benchmarks should move to the float set.

dbeyer commented:
> What about #581? I also think these benchmarks should move to the float set.

I agree. @1933131, could you please change the folder/set-file accordingly?

dbeyer commented:
@1933131 Any update?

1933131 commented:
I have done all the changes requested.
1) Changing int to unsigned int to prevent signed overflow.
2) Updated the folder name and added to floats category. Ideally the benchmarks require loop analysis, combined with floating point calculations.

dbeyer commented:
Did you see the merge conflict that needs to get resolved?

1933131 commented:
I saw the merger conflicts. It was due to addition of benchmarks in Reachsafety-float.set category, and I didn't update my forked repository, before commit. Hence, the issue came up. I fixed it, now.
